MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -216,67 +217,234 @@ namespace WPEFramework {
                         system(cmd.c_str());
                     }
                 }
             }
-            /* Here in Solicited we start with RFC so no
+            /* Here in Solicited, we start with RFC so no
              * need to wait for any DCM events */
             else if( SOLICITED_MAINTENANCE == g_maintenance_type && internetConnectStatus){
-                    LOGINFO("=============SOLICITED_MAINTENANCE===============");
-                    cmd=task_names_foreground[0].c_str();
-                    cmd+=" &";
-                    cmd+="\0";
-                    m_task_map[task_names_foreground[0].c_str()]=true;
-                    LOGINFO("Starting Script (SM) :  %s \n", cmd.c_str());
-                    system(cmd.c_str());
-                    cmd="";
-                    for (i=1;i<task_count;i++){
-                        task_thread.wait(lck);
-                        cmd=task_names_foreground[i].c_str();
-                        cmd+=" &";
-                        cmd+="\0";
-                        m_task_map[task_names_foreground[i].c_str()]=true;
-                        LOGINFO("Starting Script (SM) :  %s \n", cmd.c_str());
-                        if (!m_abort_flag){
-                            system(cmd.c_str());
-                        }
+                LOGINFO("=============SOLICITED_MAINTENANCE===============");
+                cmd = tasks[0];
+                cmd += " &";
+                cmd += "\0";
+                m_task_map[tasks[0]] = true;
+                LOGINFO("Starting Script (SM) :  %s \n", cmd.c_str());
+                system(cmd.c_str());
+                cmd="";
+                for( i = 1; i < tasks.size(); i++){
+                    LOGINFO("Waiting to unlock.. [%d/%d]",i,tasks.size());
+                    task_thread.wait(lck);
+                    cmd = tasks[i];
+                    cmd += " &";
+                    cmd += "\0";
+                    m_task_map[tasks[i]]=true;
+                    if ( !m_abort_flag ){
+                        LOGINFO("Starting Script (SM) :  %s \n",cmd.c_str());
+                        system(cmd.c_str());
                     }
+                }
             }
-
             m_abort_flag=false;
             LOGINFO("Worker Thread Completed");
-            if (false == internetConnectStatus) {
+            if ( false == internetConnectStatus ) {
                 MaintenanceManager::_instance->onMaintenanceStatusChange(MAINTENANCE_ERROR);
                 LOGINFO("Maintenance completed as it is offline mode");
             }
         }
 
+        const string MaintenanceManager::getActivatedStatus()
+        {
+            JsonObject joGetParams;
+            JsonObject joGetResult;
+            std::string callsign = "org.rdk.AuthService.1";
+            std::string token;
+            uint8_t i = 0;
+            bool isAuthSerivcePluginActive = false;
+#if defined(BUILD_ENABLE_NOT_ACTIVATED)
```

Comment:
Pls remove these default values. These are internal to getActivationStatus and might change in the future. If we cannot get the activation status after 4 retries, then can we switch to existing behavior and proceed with IsDeviceOnline check.

Reply:
removed.